std/assembly/atomics.ts

```diff
@@ -0,0 +1,89 @@
+import { ArrayBufferView } from "./arraybuffer";
+
+export namespace Atomics {
+
+  // @ts-ignore: decorator
+  @inline
+  export function load<T>(array: ArrayBufferView, index: i32): T {
```

Comment:
Wondering if there is a way to make this behave like in JS, that is without having to provide a type argument (`T` cannot be inferred). Maybe something along the lines of

```ts
function load<T>(array: T extends ArrayBufferView, index: i32): valueof<T> {
  // use valueof<T> etc. ?
}
```

Reply:
Yeah. Nice catch!

Reply:
Same can be done for most of the other methods I believe :)

Reply:
I guess it's not necessary because rest have input value with type `T` except `notify` which I already enchanced

Reply:
However this little different from. Hmm I think I convert rest as well